backends/kv/reader.go

```diff
@@ -212,6 +220,65 @@ func init() {
 	}
 }
 
+func (kv *Backend) getPartitions(path string, container v3io.Container, marker string) ([]string, error) {
+	var partitions []string
+	input := &v3io.GetContainerContentsInput{Path: path, DirectoriesOnly: true, Marker: marker}
+	res, err := container.GetContainerContentsSync(input)
+	if err != nil {
+		return nil, err
+	}
+	out := res.Output.(*v3io.GetContainerContentsOutput)
+	if len(out.CommonPrefixes) > 0 {
+		for _, partition := range out.CommonPrefixes {
+			partitions = append(partitions, partition.Prefix)
+			parts, err := kv.getPartitions(partition.Prefix, container, "")
+			if err != nil {
+				return nil, err
+			}
+			partitions = append(partitions, parts...)
+		}
+	}
+
+	if out.NextMarker != "" {
+		parts, err := kv.getPartitions(path, container, out.NextMarker)
```

Comment:
You should swap this recursion out for a loop, to avoid a potential stack overflow when a table has many partitions.